R/mergelist.R

```diff
@@ -0,0 +1,31 @@
+cbindlist = function(x, copy=TRUE) {
+  ans = .Call(Ccbindlist, x, copy)
+  setDT(ans)
+  ans
+}
+
+mergelist = function(x, on, how) {
+  # note that 'mult' is fixed to "first"
+  # expand or subset each DT by joining to complete columns set specified in 'on'
+  # now all DT share the same order and row count so we can safely stack them by cbind
+  stopifnot(is.list(x))
+  if (!length(x)) return(x)
+  stopifnot(vapply(x, is.data.table, FALSE))
+  if (length(x)<2L) return(copy(x))
+  stopifnot(is.character(on), vapply(x, function(x) all(on %chin% names(x)), FALSE))
```

Comment:
reusing `x` to mean different things in the same line is a bit hard to read... `function(xi)` maybe?

Reply:
to goal is to move it to C actually